modules/basics/src/main/java/com/opengamma/strata/basics/value/ValueSchedule.java

```diff
@@ -102,50 +122,83 @@ public static ValueSchedule of(double initialValue, ValueStep... steps) {
    * @return the value schedule
    */
   public static ValueSchedule of(double initialValue, List<ValueStep> steps) {
-    return new ValueSchedule(initialValue, ImmutableList.copyOf(steps));
+    return new ValueSchedule(initialValue, ImmutableList.copyOf(steps), null);
+  }
+
+  /**
+   * Obtains an instance from an initial value and a sequence of steps.
+   * <p>
+   * The sequence defines changes from one date to another date using a frequency.
+   * For example, the value might change every year from 2011-06-01 to 2015-06-01.
+   * 
+   * @param initialValue  the initial value used for the first period
+   * @param stepSequence  the full definition of how the value changes over time
+   * @return the value schedule
+   */
+  public static ValueSchedule of(double initialValue, ValueStepSequence stepSequence) {
+    return new ValueSchedule(initialValue, ImmutableList.of(), stepSequence);
   }
 
   //-------------------------------------------------------------------------
   /**
    * Resolves the value and adjustments against a specific schedule.
    * <p>
    * This converts a schedule into a list of values, one for each schedule period.
-   * <p>
-   * The output list is immutable and matches the input list.
-   * Use {@link Doubles#toArray} to efficiently access the list as a {@code double[]}
-   * (without breaking encapsulation or immutability).
    * 
    * @param periods  the list of schedule periods
    * @return the values, one for each schedule period
+   * @deprecated Use {@link #resolveValues(Schedule)}
    */
+  @Deprecated
   public List<Double> resolveValues(List<SchedulePeriod> periods) {
+    return resolveValues(periods, RollConventions.NONE).toList();
+  }
+
+  /**
+   * Resolves the value and adjustments against a specific schedule.
+   * <p>
+   * This converts a schedule into a list of values, one for each schedule period.
+   * 
+   * @param schedule  the schedule
+   * @return the values, one for each schedule period
+   */
+  public DoubleArray resolveValues(Schedule schedule) {
+    return resolveValues(schedule.getPeriods(), schedule.getRollConvention());
+  }
+
+  // resolve the values
+  private DoubleArray resolveValues(List<SchedulePeriod> periods, RollConvention rollConv) {
+    // handle simple case where there are no steps
```

Comment:
This comment looks like it's in the wrong place. Like you did a refactor that flipped the sense of the `if`.